owners/test/ownership/parser.test.js

```diff
@@ -245,105 +259,102 @@ describe('owners parser', () => {
   describe('parseRuleDefinition', () => {
     describe('when given something not a rule definition', () => {
       it('reports an error', () => {
-        const {errors} = parser._parseRuleDefinition('', 'NOT A RULE DEF');
-        expect(errors[0].message).toEqual(
-          'Expected rule definition; got string'
-        );
+        const {errors} = parseRuleDefinition('NOT A RULE DEF');
+        expect(errors[0].message).toEqual('`OWNERS.rules[0]` should be object');
       });
 
       it('returns no result', () => {
-        const {result} = parser._parseRuleDefinition('', 'NOT A RULE DEF');
-        expect(result).toBeUndefined();
+        const {rule} = parseRuleDefinition('NOT A RULE DEF');
+        expect(rule).toBeUndefined();
       });
     });
 
     describe('when an unexpected key is present', () => {
       it('reports an error', () => {
-        const {errors} = parser._parseRuleDefinition('', {
+        const {errors} = parseRuleDefinition({
           owners: [{name: 'blah'}],
           notify: true, // This should be in the owner, not the rule
         });
         expect(errors[0].message).toEqual(
-          'Unexpected key(s) "notify" in rule definition'
+          '`OWNERS.rules[0]` should NOT have additional properties'
```

Comment:
nit: is it possible to set custom error messages in the schema so that the context stays?

Reply:
No :disappointed: The best I could do is post-process the error message, which would add almost as much logic as this PR removes. I definitely dislike that aspect; a lot of clarity is lost when something is invalid